src/parameterFiles/DeploymentFileMapping.ts

```diff
@@ -0,0 +1,120 @@
+                        let resolvedPath = path.resolve(path.dirname(templateFileUri.fsPath), paramFiles[fileNameKey]);
+
+                        // If the user has an entry in both workspace and user settings, vscode combines the two objects,
+                        //   with workspace settings overriding the user settings.
+                        // If there are two entries differing only by case, allow the last one to win, because it will be
+                        //   the workspace setting value
+                        paramFile = !!resolvedPath ? Uri.file(resolvedPath) : undefined;
+                    }
+                }
+            }
+
+            return paramFile;
+        }
+
+        return undefined;
+    }
+
+    //asdf
+    public getTemplateFile(parameterFileUri: Uri): Uri | undefined {
+        const paramFiles: { [key: string]: string } | undefined =
+            this.configuration.get<{ [key: string]: string }>(configKeys.parameterFiles)
+            // tslint:disable-next-line: strict-boolean-expressions
+            || {};
+        if (typeof paramFiles === 'object') {
+            const normalizedTargetParamPath = normalizePath(parameterFileUri.fsPath);
+            let templateFile: Uri | undefined;
+
+            // Can't do a simple lookup because need to be case-insensitivity tolerant on Win32
+            for (let fileNameKey of Object.getOwnPropertyNames(paramFiles)) {
+                const paramFileName = paramFiles[fileNameKey]; // asdf can this be undefined?
+                if (typeof paramFileName !== "string") {
+                    continue;
+                }
+
+                // Resolve relative to template file's folder
+                let resolvedPath = path.resolve(path.dirname(parameterFileUri.fsPath), paramFileName);
+                const normalizedParamPath: string = normalizePath(resolvedPath);
+
+                // If the user has an entry in both workspace and user settings, vscode combines the two objects,
+                //   with workspace settings overriding the user settings.
+                // If there are two entries differing only by case, allow the last one to win, because it will be
+                //   the workspace setting value
+
+                if (normalizedParamPath === normalizedTargetParamPath) {
+                    templateFile = !!fileNameKey ? Uri.file(fileNameKey) : undefined;
+                }
+            }
+
+            return templateFile;
+        }
+
+        return undefined;
+    }
+
+    /**
+     * Sets a mapping from a template file to a parameter file
+     */
+    public async mapParameterFile(templateUri: Uri, paramFileUri: Uri | undefined): Promise<void> {
+        const relativeParamFilePath: string | undefined = paramFileUri ? getFriendlyPathToParameterFile(templateUri, paramFileUri) : undefined;
+        const normalizedTemplatePath = normalizePath(templateUri.fsPath);
+
+        // We only want the values in the user settings
```

Comment:
>We only want the values in the user settings [](start = 11, length = 44)

So we will never save to workspace settings? What about the comments above talking about settings getting merged from workspace settings and user settings? How do things get into workspace settings?

Reply:
The user can always add the settings to the workspace, but we only want to update them in the user settings.  vscode automatically merges the two, though.